app/uk/gov/hmrc/gform/gform/processor/FormProcessor.scala

```diff
@@ -151,8 +170,13 @@ class FormProcessor(
         EnteredVariadicFormData.empty,
         true
       ) { _ => _ => maybeSectionNumber =>
+        val pageIdToRemove = updFormModel.brackets.addToListBracket(addToListId).source.pageIdToDisplayAfterRemove
+        val pageIdSectionNumberMap = updFormModel.pageIdSectionNumberMap
         val sectionNumber =
-          if (isLastIteration)
+          if (isLastIteration && pageIdToRemove.isDefined) {
+            computePageLink(pageIdToRemove.get, pageIdSectionNumberMap)
```

Comment:
Also please refactor this `.isDefined` followed by `.get` pattern

Reply:
Please remove unsafe `.get` calls. Something like this should suffice:

```suggestion
        val sectionNumber =
          if (isLastIteration) {
            pageIdToRemove.fold(
              maybeSectionNumber
                .map(addToListBracket.iterationForSectionNumber(_).firstSectionNumber)
                .getOrElse(sn)
            ) { pageId =>
              computePageLink(pageId, pageIdSectionNumberMap)
                .getOrElse(throw new Exception(s"Unable to find section number for pageId: $pageId"))
            }
          } else
            sn
```